### PR TITLE
Bundle status refresh endpoint refreshes only the specified bundle 

### LINKED ISSUE #1999: Bundle status refresh endpoint refreshes the entire worksheet
It should refresh just the particular bundle IDs specified in the query string.

### PR DESCRIPTION
**Pull request by yuqijin**
### Reasons for making this change

We had a `partial update` mechanism in our system, however, it seems like this mechanism lost consistency between backend and frontend during the development, so even if a bundle status refresh endpoint always refreshes the entire worksheet. Now the bugs are fixed and only the bundles that need updating aren't `None` during network transmission.

### Related issues

fixes #1999 

### Screenshots
Before: 
Screen Shot 2021-01-24 at 6 54 41 PM

After:
Screen Shot 2021-01-24 at 6 53 16 PM
Screen Shot 2021-01-24 at 6 47 10 PM

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: codalab/rest/interpret.py [hunk 1]
```diff
@@ -300,17 +300,17 @@ def fetch_interpreted_worksheet(uuid):
     # The bundle_infos for bundles that don't need updating are also None.
     if bundle_uuids:
         for i, block in enumerate(interpreted_blocks['blocks']):
```
Comment (epicfaace) [inline codalab/rest/interpret.py:302]: This looks good. Could we add a test within the `test/unit/rest` directory that ensures that the partial refresh endpoint never regresses again?

Comment (yuqijin) [reply codalab/rest/interpret.py:302]: Sure. The test has been added

File: codalab/rest/interpret.py [hunk 2]
```diff
@@ -347,7 +347,7 @@ def fetch_interpreted_worksheet(uuid):
     # Frontend doesn't use individual 'items' for now
     del worksheet_info['items']
     if bundle_uuids:
-        return {'blocks': worksheet_info['blocks']}
+        return {'blocks': worksheet_info['blocks'], 'uuid': uuid}
```
Comment (epicfaace) [inline codalab/rest/interpret.py:350]: What is `uuid` equal to? I thought `bundle_uuids` can specify multiple bundle uuids?

File: codalab/rest/interpret.py [hunk 3]
```diff
@@ -300,17 +300,17 @@ def fetch_interpreted_worksheet(uuid):
     # The bundle_infos for bundles that don't need updating are also None.
     if bundle_uuids:
         for i, block in enumerate(interpreted_blocks['blocks']):
-            if 'bundle_info' not in block:
+            if not ('bundles_spec' in block and 'bundle_infos' in block['bundles_spec']):
```
Comment (epicfaace) [inline codalab/rest/interpret.py:303]: Can you explain a bit what your code changes do here? Maybe just in the PR description / a comment here

File: frontend/src/components/worksheets/Worksheet/Worksheet.js [hunk 4]
```diff
@@ -1524,9 +1524,26 @@ class Worksheet extends React.Component {
         } else {
             var ws = _.clone(this.state.ws);
             for (var i = 0; i < partialUpdateItems.length; i++) {
-                if (!partialUpdateItems[i]) continue;
+                if (
+                    !partialUpdateItems[i] ||
+                    !(
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet/Worksheet.js:1532]: Can you add a comment explaining what this is looking for?

File: codalab/rest/interpret.py (hunk 2 above)
Comment (yuqijin) [reply codalab/rest/interpret.py:350]: `uuid` refers to the worksheet's uuid. `bundle_uuids` refers to the bundles' uuids.

File: codalab/rest/interpret.py (hunk 3 above)
Comment (yuqijin) [reply codalab/rest/interpret.py:303]: It is used to check whether the block contains the field of 'bundle_infos' to determine whether it is a non-None block containing a list of bundles.

File: frontend/src/components/worksheets/Worksheet/Worksheet.js (hunk 4 above)
Comment (yuqijin) [reply frontend/src/components/worksheets/Worksheet/Worksheet.js:1532]: Added!

**Comment by yuqijin** [comment]
Also fixes #3208 that not all the bundle id will be included in the GET URL:

Screen Shot 2021-02-02 at 10 17 54 PM

File: codalab/rest/interpret.py [hunk 5]
```diff
@@ -300,17 +300,17 @@ def fetch_interpreted_worksheet(uuid):
...
                 interpreted_blocks['blocks'][i] = None
             else:
-                if isinstance(block['bundle_info'], dict):
-                    block['bundle_info'] = [block['bundle_info']]
+                if isinstance(block['bundles_spec']['bundle_infos'], dict):
+                    block['bundles_spec']['bundle_infos'] = [block['bundles_spec']['bundle_infos']]
```
Comment (epicfaace) [inline codalab/rest/interpret.py:307]: Why did you change `block['bundle_info']` to `block['bundles_spec']['bundle_infos']`? Can each block now contain multiple bundle infos (which wasn't possible before)?

Comment (yuqijin) [reply codalab/rest/interpret.py:307]: Because the object `block` only contains the field of `bundles_spec`, I believe it is because the code has not been updated during development. 
And yes, each block can contain multiple bundle Infos, since a block can be a table, then it will make sense.

**Review by epicfaace — APPROVED** [review]
